modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/MustacheScriptEngineService.java

```diff
@@ -85,8 +88,21 @@ public MustacheScriptEngineService(Settings settings) {
      * */
     @Override
     public Object compile(String template, Map<String, String> params) {
-        /** Factory to generate Mustache objects from. */
-        return (new JsonEscapingMustacheFactory()).compile(new FastStringReader(template), "query-template");
+        boolean escapeJson = true;
+        String escapeJsonStr = params.get(ESCAPE_JSON_PARAM);
```

Comment:
I don't think that the scripting/template service/engine should be json oriented by default. I think it should work with potentially any content type (json, yaml, plain text, or whatever). I wouldn't have a parameter called `escape_son`, but instead maybe have a `content_type` parameter? and expect the values of this to be actual content types (i.e. `text/plain`, `application/json`, `application/yaml`, etc...)? By default we should simply assume `text/plain` and `utf8`. Then it'll be the responsibility of the client of this service to provide the context (e.g. the search templates will add `content_type=application/json`).